client/landing/domains/registrant-verification/index.jsx

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DomainsLandingHeader from '../header';
+import DomainsLandingContentCard from '../content-card';
+import { CALYPSO_CONTACT } from 'lib/url/support';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class RegistrantVerificationPage extends Component {
+	static propTypes = {
+		domain: PropTypes.string.isRequired,
+		email: PropTypes.string.isRequired,
+		token: PropTypes.string.isRequired,
+	};
+
+	state = {
+		isLoading: true,
+		success: false,
+		error: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = this.getLoadingState();
+	}
+
+	componentWillMount() {
+		const { domain, email, token } = this.props;
+		wpcom.domainsVerifyRegistrantEmail( domain, email, token ).then(
+			() => {
+				this.setState( this.getVerificationSuccessState() );
+			},
+			error => {
+				this.setErrorState( error );
+			}
+		);
+	}
+
+	getLoadingState = () => {
+		const { translate } = this.props;
+		return {
+			isLoading: true,
+			illustration: '/calypso/images/illustrations/whoops.svg',
+			title: translate( 'Verifying your contact information…' ),
+			message: 'Loading…',
+			actionTitle: null,
+			actionCallback: null,
+			footer: 'Loading…',
+		};
+	};
+
+	getVerificationSuccessState = () => {
+		const { domain, translate } = this.props;
+		return {
+			illustration: '/calypso/images/illustrations/illustration-ok.svg',
+			title: translate( 'Success!' ),
+			message: translate(
+				'Thank your for verifying your contact information for:{{br /}}{{strong}}%(domain)s{{/strong}}.',
+				{
+					args: {
+						domain: domain,
+					},
+					components: {
+						strong: <strong />,
+						br: <br />,
+					},
+				}
+			),
+			actionTitle: null,
+			actionCallback: null,
+			footer: translate( 'All done. You can close this window now.' ),
+			isLoading: false,
+		};
+	};
+
+	getExpiredState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate(
+				'This email has expired.{{br/}}Please resend the verification email and try again.',
+				{
+					components: {
+						br: <br />,
+					},
+				}
+			),
+			actionTitle: translate( 'Resend Email' ),
+			actionCallback: this.resendEmail,
+		};
+	};
+
+	getEmailMismatchState = () => {
+		const { domain, email, translate } = this.props;
+
+		return {
+			message: translate(
+				'{{strong}}%(email)s{{/strong}} is different from the email we have on record for {{strong}}%(domain)s{{/strong}}.',
```

Comment:
Maybe let's add a sentence on what to do next? Otherwise, it feels like we're just leaving the user to figure it out themselves.

AFAICT, this state is triggered when the user clicks an old link, so maybe let's nudge the user to check their email for a newer email, etc.